devito/dse/rewriters.py

```diff
@@ -406,6 +408,47 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        A pass to perform only the basic skewing in order to have valid data
+        dependences.
+        """
+        skew_factor = configuration['skew_factor']
+        skew_dim, mapper, int_mapper = None, {}, []
```

Comment:
Exercise: rethink this entire piece of code and make it:
- more general (see also comments below)
- more robust
- **simpler** (e.g., here you're using two dicts, one list, one integer counter, and some other stuff...to do something which is very basic) ...
- ... and consequently easier to read

use as model the other algorithms in devito. 

This is a coding exercise

recall: let **simplicity** guide you. If it's not simple, there's a good chance it's flawed/wrong/broken/ etc.